keras_cv/models/segmentation/segment_anything/sam.py

```diff
@@ -203,6 +195,15 @@ def __init__(self, *, backbone, prompt_encoder, mask_decoder, **kwargs):
         self.prompt_encoder = prompt_encoder
         self.mask_decoder = mask_decoder
 
+    # TODO(ianstenbit): Do something more elegant to handle empty prompts.
+    def predict_step(self, *args, **kwargs):
```

Comment:
I don't think this needs to be done at each predict step. I was suggesting something like:

```suggestion
    def predict(self, inputs, ...):
        B = self._get_batch_size(inputs)
        points = inputs.get("points", ops.empty(B, 0, 2))
        ...
        inputs = {
            "points": points,
            ...
        }
        super().predict(inputs, ...)
```

Any reason why override the `predict_step` instead? I think it'd add some overhead at each step.

Reply:
One more thing: the `call` method will also need to be updated since it also expects a full prompt dictionary.

Reply:
Your proposed solution won't work well when the inputs are not just a single dictionary (e.g. if they're a `tf.data.Dataset` with many batches), so I used `predict_step`

And yes, `call` still requires all the arguments. I think the only workaround for this is to no longer use a functional subclass model but instead a pure subclass model, which is probably what we should do in the long run. For now I was just trying to get the `predict` API working. Before 0.7 is cut we should resolve this.

Reply:
> Your proposed solution won't work well when the inputs are not just a single dictionary (e.g. if they're a `tf.data.Dataset` with many batches), so I used `predict_step`

Ah, that makes sense. Thanks for clarifying!

> And yes, `call` still requires all the arguments. I think the only workaround for this is to no longer use a functional subclass model but instead a pure subclass model, which is probably what we should do in the long run. For now I was just trying to get the `predict` API working. Before 0.7 is cut we should resolve this.

Does Keras compile the model if the `call` method is overridden?